hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -1488,4 +1488,23 @@ public static Comparable<?> unwrapAvroValueWrapper(Object avroValueWrapper) {
     }
   }
 
+  public static boolean containsUnsupportedTypesForFileGroupReader(Schema schema) {
+    switch (schema.getType()) {
+      case RECORD:
+        for (Field field : schema.getFields()) {
+          if (containsUnsupportedTypesForFileGroupReader(field.schema())) {
+            return true;
+          }
+        }
+        return false;
+      case ARRAY:
+        return containsUnsupportedTypesForFileGroupReader(schema.getElementType());
+      case MAP:
+        return containsUnsupportedTypesForFileGroupReader(schema.getValueType());
+      case UNION:
+        return containsUnsupportedTypesForFileGroupReader(getActualSchemaFromUnion(schema, null));
+      default:
+        return schema.getType() == Schema.Type.ENUM || schema.getLogicalType() instanceof LogicalTypes.Decimal;
```

Comment:
Decimal type is very common, we need to support it for fg reader.

Reply:
Now it's supported by fixing the Spark Parquet writer.